packages/smooth_app/lib/cache/files/files_cache_impl.dart

```diff
@@ -0,0 +1,121 @@
+// ignore_for_file: avoid_classes_with_only_static_members
+// ignore_for_file: avoid_slow_async_io
+part of 'files_cache.dart';
+
+/// Implementation based on [getTemporaryDirectory]
+class _ShortLivingFilesCacheManagerImpl extends _FilesCacheManagerImpl {
+  _ShortLivingFilesCacheManagerImpl._() : super._();
+
+  @override
+  Future<void> _init(String name) async {
+    return _initCache(name, await getTemporaryDirectory());
+  }
+}
+
+/// Implementation based on [getApplicationDocumentsDirectory]
+class _LongLivingFilesCacheManagerImpl extends _FilesCacheManagerImpl {
+  _LongLivingFilesCacheManagerImpl._() : super._();
+
+  @override
+  Future<void> _init(String name) async {
+    return _initCache(name, await getApplicationDocumentsDirectory());
+  }
+}
+
+abstract class _FilesCacheManagerImpl extends FilesCache {
+  factory _FilesCacheManagerImpl(FilesCacheType type) {
+    switch (type) {
+      case FilesCacheType.shortLiving:
+        return _ShortLivingFilesCacheManagerImpl._();
+      case FilesCacheType.longLiving:
+        return _LongLivingFilesCacheManagerImpl._();
+    }
+  }
+
+  _FilesCacheManagerImpl._() : super();
+
+  static const String _DEFAULT_FOLDER_NAME = 'default';
+  late Directory _directory;
+
+  Future<void> _init(String name);
+
+  /// Please never call this method call directly, but use instead [_init].
+  Future<void> _initCache(String name, Directory directory) async {
+    _directory = Directory(
+      join(
+        directory.absolute.path,
+        'temporary_files_cache',
```

Comment:
Always `'temporary_files_cache'`?

Reply:
Yes, I want to be sure to have a subfolder containing the cache.
It will be `folder_name/temporary_files_cache/xxxx`
Removing `temporary_files_cache` could potentially interfere with existing directories

Reply:
Ah ok 😅, it was just for the name of the folder!
Sorry I've misunderstood.